File: backend/workspaces/sessions/default-candidate/cpp-task-manager/task_manager.cpp
#include <vector>
#include <string>
#include <algorithm>

struct Task {
    int id;
    std::string title;
    std::string priority;
    bool done;
    Task(int i, const std::string& t, const std::string& p) : id(i), title(t), priority(p), done(false) {}
};

class TaskManager {
private:
    std::vector<Task> tasks;
    int nextId = 1;

public:
    Task addTask(const std::string& title, const std::string& priority) {
        // TODO: Create a new Task, add to tasks vector, increment nextId
        // Return the created task
        return Task(0, "", "");
    }

    void completeTask(int taskId) {
        // TODO: Find task by id and set done = true
    }

    void removeTask(int taskId) {
        // TODO: Remove task by id from tasks vector
    }

    std::vector<Task> listTasks() {
        // TODO: Return all tasks
        return {};
    }

    std::vector<Task> filterByPriority(const std::string& priority) {
        // TODO: Return tasks matching the given priority
        return {};
    }
};

File: backend/workspaces/sessions/test-java-1/TaskManager.java
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class TaskManager {
    private List<Task> tasks = new ArrayList<>();
    private int nextId = 1;

    public Task addTask(String title, String priority) {
        // TODO: Create a new Task, add to tasks list, increment nextId
        // Return the created task
        return null;
    }

    public void completeTask(int taskId) {
        // TODO: Find task by id and set done = true
    }

    public void removeTask(int taskId) {
        // TODO: Remove task by id from tasks list
    }

    public List<Task> listTasks() {
        // TODO: Return all tasks
        return null;
    }

    public List<Task> filterByPriority(String priority) {
        // TODO: Return tasks matching the given priority
        return null;
    }
}
